client/Command/ListSnapshots/Options.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) [2019] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact Novell, Inc.
+ *
+ * To contact Novell about this file by physical or electronic mail, you may
+ * find current contact information at www.novell.com.
+ */
+
+#ifndef SNAPPER_CLI_COMMAND_LIST_SNAPSHOTS_OPTIONS_H
+#define SNAPPER_CLI_COMMAND_LIST_SNAPSHOTS_OPTIONS_H
+
+#include "client/Options.h"
+#include "client/GlobalOptions.h"
+#include "client/Command/ColumnsOption.h"
+#include "client/Command/ListSnapshots.h"
+#include "client/Command/ListSnapshots/SnappersData.h"
+
+namespace snapper
+{
+    namespace cli
+    {
+
+	class Command::ListSnapshots::Options : public cli::Options
+	{
+
+	public:
+
+	    struct Columns
+	    {
+		static const std::string CONFIG;
+		static const std::string SUBVOLUME;
+		static const std::string NUMBER;
+		static const std::string DEFAULT;
+		static const std::string ACTIVE;
+		static const std::string TYPE;
+		static const std::string DATE;
+		static const std::string USER;
+		static const std::string USED_SPACE;
+		static const std::string CLEANUP;
+		static const std::string DESCRIPTION;
+		static const std::string USERDATA;
+		static const std::string PRE_NUMBER;
+		static const std::string POST_NUMBER;
+		static const std::string POST_DATE;
+	    };
```

Comment:
Can this be done with an enum?

Reply:
I was evaluating it, but I decided to directly use string constants at the end for some reasons:

* I would like to continue using simply strings in the `ColumnsOption` class, so with enums we would have to define the string representation of each enum value anyway.

* An string representation of each column is also needed to check if the column was selected via the *--columns* option.

* We could use the `Snapper/Enum.h` class to covert enum values to strings and vice-versa, but it implies to define the enum outside each `Option` class. I would prefer to have `snapper::cli::ListSnapshots::Options::Columns::NUMBER` instead of `snapper::ListSnapshotsColumns::NUMBER` (I don't like to populate *snapper* namespace with those enums).

For these reasons, I decided to directly have a set of columns names (which can be easily referenced by a variable like `CommandName::Options::Columns::COLUMN`) instead of an enum with its conversion to strings.